src/rpc/rpc.cc

```diff
@@ -295,5 +299,129 @@ DGL_REGISTER_GLOBAL("distributed.server_state._CAPI_DGLRPCGetServerState")
   *rv = st;
 });
 
+//////////////////////////// KVStore ////////////////////////////
+
+DGL_REGISTER_GLOBAL("distributed.rpc._CAPI_DGLRPCGetLocalPartitionData")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+  NDArray ID = args[0];
+  NDArray part_id = args[1];
+  int local_machine_id = args[2];
+  int64_t* ID_data = static_cast<int64_t*>(ID->data);
+  int64_t* part_id_data = static_cast<int64_t*>(part_id->data);
+  int64_t ID_size = ID.GetSize() / sizeof(int64_t);
+  std::vector<int64_t> global_id;
+  for (int64_t i = 0; i < ID_size; ++i) {
+    if (part_id_data[i] == local_machine_id) {
+      global_id.push_back(ID_data[i]);
+    }
+  }
+  NDArray res_tensor = dgl::aten::VecToIdArray<int64_t>(global_id);
+  *rv = res_tensor;
+});
+
+DGL_REGISTER_GLOBAL("distributed.rpc._CAPI_DGLRPCFastPull")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+  // Input
+  std::string name = args[0];
+  int local_machine_id = args[1];
+  int machine_count = args[2];
+  int group_count = args[3];
+  int client_id = args[4];
+  int service_id = args[5];
+  int msg_seq = args[6];
+  std::string pickle_data = args[7];
+  NDArray ID = args[8];
+  NDArray part_id = args[9];
+  NDArray g2l_id = args[10];
+  NDArray local_data = args[11];
+  // Data
+  int64_t ID_size = ID.GetSize() / sizeof(int64_t);
+  int64_t* ID_data = static_cast<int64_t*>(ID->data);
+  int64_t* part_id_data = static_cast<int64_t*>(part_id->data);
+  int64_t* g2l_id_data = static_cast<int64_t*>(g2l_id->data);
```

Comment:
btw, why not use `dgl_id_t`, instead of int64_t?

Reply:
this `g2l_id_data` is very confusing. My understanding is that this is the global Ids of the nodes in the local partition.

Reply:
I changed all the `int64_t` to `dgl_id_t`